conan/internal/api/install/generators.py

```diff
@@ -118,18 +118,23 @@ def write_generators(conanfile, app):
             with conanfile_exception_formatter(conanfile, "generate"):
                 conanfile.generate()
 
-    if conanfile.virtualbuildenv:
-        mkdir(new_gen_folder)
-        with chdir(new_gen_folder):
-            from conan.tools.env.virtualbuildenv import VirtualBuildEnv
-            env = VirtualBuildEnv(conanfile)
-            env.generate()
-    if conanfile.virtualrunenv:
-        mkdir(new_gen_folder)
-        with chdir(new_gen_folder):
-            from conan.tools.env import VirtualRunEnv
-            env = VirtualRunEnv(conanfile)
-            env.generate()
+    if envs_generation is None:
+        envs_generation = "auto"
```

Comment:
Not sure about this line. This might simplify chekcing when we add a new not_empty function, but maybe for now it's cleaner to have the following check also check for none instead?

I have allowed for `auto` to be the same as the default `None` so that it shows up nicely in the command help
```
  --envs-generation {auto,never}
                        Generation strategy for virtual environment files for the root
```
And it being None by default would be the same as typing "auto", as mentioned in https://github.com/conan-io/conan/pull/16935#discussion_r1744014494

Reply:
Still not very clear why a ``--envs-generatior=auto`` that is equal to doing nothing is needed, maybe the ``auto`` value can be dropped?